tests/app/main/views/test_communications.py

```diff
@@ -59,35 +57,62 @@ def test_ccs_sourcing_can_not_post_documents_for_framework(self, data_api_client
         # check we are mocking s3 as we must not actually post there!
         assert isinstance(s3.S3, mock.Mock)
 
-        data_api_client.get_frameworks.return_value = {"frameworks": []}
         self.user_role = "admin-ccs-sourcing"
-        framework_slug = self.load_example_listing('framework_response')['frameworks']['slug']
-        dummy_file = BytesIO(u'Lorem ipsum dolor sit amet'.encode('utf8'))
 
         response = self.client.post(
-            "/admin/communications/{}".format(framework_slug),
+            "/admin/communications/{}".format(self.framework_slug),
             data={
-                'communication': (dummy_file, 'test-comm.pdf'),
-                'clarification': (dummy_file, 'test-clar.pdf'),
+                'communication': (self.dummy_file, 'test-comm.pdf'),
+                'clarification': (self.dummy_file, 'test-clar.pdf'),
             }
         )
         assert response.status_code == 403
 
     def test_post_bad_documents_for_framework(self, data_api_client):
-        data_api_client.get_frameworks.return_value = {"frameworks": []}
-
-        framework_slug = self.load_example_listing('framework_response')['frameworks']['slug']
-        dummy_file = BytesIO(u'Lorem ipsum dolor sit amet'.encode('utf8'))
 
         self.client.post(
-            "/admin/communications/{}".format(framework_slug),
+            "/admin/communications/{}".format(self.framework_slug),
             data={
-                'communication': (dummy_file, 'test-comm.unknown'),
-                'clarification': (dummy_file, 'test-clar.unknown'),
+                'communication': (self.dummy_file, 'test-comm.unknown'),
+                'clarification': (self.dummy_file, 'test-clar.unknown'),
             }
         )
 
         flash_messages = self._get_flash_messages()
         # definitely a misunderstanding here about what an appropriate 'message category' looks like
         assert flash_messages.get('not_open_document_format_or_csv') == 'communication'
         assert flash_messages.get('not_pdf') == 'clarification'
+
+    def test_communications_file_saves_with_correct_path(self, data_api_client):
+        assert isinstance(s3.S3, mock.Mock)
+
+        response = self.client.post(
+            "/admin/communications/{}".format(self.framework_slug),
+            data={
+                'communication': (self.dummy_file, 'test-comm.pdf'),
+            }
+        )
+
+        assert response.status_code == 302
+        self.s3.return_value.save.assert_called_once_with(
+            'g-cloud-8/communications/updates/communications/test-comm.pdf',
+            mock.ANY,
```

Comment:
An inline comment here saying what this would make this really readable
i.e.

```mock.ANY,  # this is...```

Reply:
Done